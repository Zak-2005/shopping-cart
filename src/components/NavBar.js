import { Link, Outlet } from "react-router-dom";

const NavBar = (props) => {
  return (
    <>
      <ul className="navBar">
        <li>
          <Link to="/">Home</Link>
        </li>
        <li>
          <Link to="/shop/products">Shop</Link>
        </li>
        <li>
          <Link to="/cart">Cart {props.itemsInCart}</Link>
        </li>
      </ul>

      <Outlet />
    </>
  );
};
export default NavBar;
