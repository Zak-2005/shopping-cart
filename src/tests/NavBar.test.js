import { HashRouter } from "react-router-dom"
import NavBar from "../components/NavBar"
import React from "react"
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

describe("NavBar component", ()=>{
    it("Renders 3 pages ", ()=>{
        const {container} = render(<HashRouter><NavBar /></HashRouter>)
        expect(container).toMatchSnapshot();
    })
        
    }
)