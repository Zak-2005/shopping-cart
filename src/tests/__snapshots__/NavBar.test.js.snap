// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component Renders 3 pages  1`] = `
<div>
  <ul
    class="navBar"
  >
    <li>
      <a
        href="#/"
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="#/shop/products"
      >
        Shop
      </a>
    </li>
    <li>
      <a
        href="#/cart"
      >
        Cart 
      </a>
    </li>
  </ul>
</div>
`;
